### PR TITLE
introduce opt.direction

### PR DESCRIPTION
**Pull request by jakob-r**
I am not sure about the DIB, though.

### CODE REVIEW

**Comment by danielhorn** [comment]
DIB is allways mazimized

**Comment by jakob-r** [comment]
@danielhorn Can you link me a paper where the DIB is mentioned/defined?

**Comment by danielhorn** [comment]
This one could be eniough:
https://scholar.google.de/citations?view_op=view_citation&hl=de&user=WBFGirQAAAAJ&citation_for_view=WBFGirQAAAAJ:u5HHmVD_uO8C

**Comment by jakob-r** [comment]
so the name for the DIB should maybe better be "direct indicator-based search" instead of "Directed Indicator Based Search"?

**Comment by berndbischl** [comment]
> just "direct indicator-based"

thats best i think

**Comment by berndbischl** [comment]
the API is not so good and confusing now.

in the docs you say: 
- "the infill function is always minimized"
- the user can set an "opt.direction" which can also be something else (conflict with docs)
- you have getOptDirection, same name as the slot in the getter, which now returns +1 -1

this is confusing and at least the names are bad, and the docs not complete enough

**Comment by jakob-r** [comment]
getMBOInfillCriterionDirection -> getMBOInfillCritMultiplier

**Comment by berndbischl** [comment]
i did my initial review, pls ping again later when done

**Comment by berndbischl** [comment]
i guess you need to resolve quickly

will read code regardless now

**Comment by jakob-r** [comment]
wait a sec

**Comment by berndbischl** [comment]
you at least need to document internally for getInfillCritDirection why this only return 1 for minimize.
So why does it still make sense for "objective"?
Actually: does it make sense?

**Comment by berndbischl** [comment]
thinking about this: 

> you at least need to document internally for getInfillCritDirection why this only return 1 for minimize.
> So why does it still make sense for "objective"?
> Actually: does it make sense?

it seems bad. and we have initCrit now. set the objective there. and if in getDirection the setting is still "objective" throw an error

File: R/initCrit.R [hunk 1]
```diff
@@ -21,19 +21,38 @@
...
+# sets the opt.direction to minimize or maximize depending on the object function.
+initCritOptDirection = function(crit, fun) {
+  if (crit$opt.direction == "objective") {
+    ifelse
+    if (isTRUE(all(shouldBeMinimized(fun)))) {
+      crit$opt.direction = "minimize"
+    } else if (isFALSE(any(shouldBeMinimized(fun)))) {
+      crit$opt.direction = "maximize"
+    } else if (is.logical(shouldBeMinimized(fun))) {
+      stop("Don't know how to handle multi-objective functions with mixed optimization directions.")
```
Comment (jakob-r) [inline R/initCrit.R]: I feel a bit bad about this part

**Comment by berndbischl** [comment]
> I feel a bit bad about this part

yes, the code looks a bit strange. first of all, i just dislike a few style details, how ifelse, all, and isTrue is used there. 

but what is worse is, ithink: have we specified, in docs, how this works for multicrit?

**Comment by jakob-r** [comment]
Well it is all unnecessarily complicated since we have initCrit but for multiobjective we might actually want to init the infill for every component instead of at just once at the beginning. 
This would be some work but possible the problem is just that we just don't need that function `initCritOptDirection`to much that it is wort all the effort because at the moment we just use the direction in one function. Namely `renderExampleRunPlot1d`.

**Comment by jakob-r** [comment]
I checked again and all addressed points are solved.
Only the special case of multi objective optimization is not fully specified yet but imho we can merge and close.

**Pull request merged**